tests/testsuite/check_cfg.rs

```diff
@@ -336,17 +336,18 @@ fn test_false_lib() {
         .build();
 
     p.cargo("check -v")
-        .with_stderr_contains(x!("rustc" => "cfg" of "docsrs,test"))
+        .with_stderr_does_not_contain(x!("rustc" => "cfg" of "docsrs,test"))
+        .with_stderr_contains(x!("rustc" => "cfg" of "docsrs"))
```

Comment:
One way this test would silently keep passing is if we split the calls to `--check-cfg` so its `--check-cfg cfg(docsrs) --check-cfg cfg(test)`

Would really prefer testing the desired end-user result but that would mean checking for more rustc warnings.

Reply:
More generally, I'm not thrilled with these tests assertions.  They do not communicate what the expected behavior is.  This reads as if `--cfg docsrs` is being passed in.  They also rely on brittle `contains` checking which we try to avoid.

Reply:
I agree checking for those arguments is a bit brittle and the macro could use some rework (or better removal).

I will include some partial warning checking (as to not depend on the exact warning output which changes often for check-cfg/`unexpected_cfgs`). Just waiting for https://github.com/rust-lang/rust/pull/131729 to hit nightly.

Reply:
I've added the partial stderr checking for checking that the warning(s) are indeed emitted.

I have added those as the separate commit. Let me know if it isn't sufficient.